src/Compilers/Core/Portable/SourceGeneration/Nodes/DriverStateTable.cs

```diff
@@ -66,7 +95,7 @@ public NodeStateTable<T> GetLatestStateTableForNode<T>(IIncrementalGeneratorNode
                 }
 
                 // get the previous table, if there was one for this node
-                NodeStateTable<T> previousTable = _previousTable.GetStateTable(source);
+                NodeStateTable<T> previousTable = _previousTable.GetStateTableOrEmpty<T>(source);
 
                 // request the node update its state based on the current driver table and store the new result
                 var newTable = source.UpdateStateTable(this, previousTable, _cancellationToken);
```

Comment:
In the case of `ISyntaxInputNode` the `_tableBuilder[source] = newTable` line is redundant correct? The `GetSyntaxInputTable` call should already being doing the population. 

Don't think this should necessarily change but want to validate my understanding here.

Reply:
Yes, thats correct. We could potentially not add the input to `GetSyntaxInputTable` but it seems like its more effort than the double set.

---
In reply to: [632884308](https://github.com/dotnet/roslyn/pull/53383#discussion_r632884308) [](ancestors = 632884308)